specifications/api/swagger.yaml

```diff
@@ -2031,6 +2047,22 @@ paths:
               schema: *ApiPostRandomAddressData
         responses: *responsesPostRandomAddress
 
+  /byron-wallets/{walletId}/addresses/{addressId}:
+      put:
```

Comment:
Don't the REST dogmas say that PUT is for updating and POST is for creating?

Reply:
Not necessarily. `PUT` is for pushing an already created resource  assuming that a client does have the ability to create such resource entirely. This is the case here where the address pre-exists to the call, and is therefore pushed to the server. 

`POST` would be for creating a resource from partial data, or from a "constructor". There are resources where this is not possible to do differently, like the wallet. We can't possibly allow a client to "put" its own balance or sync progress, hence the POST.

Reply:
Then, for choosing between `PUT` and `PATCH`, it's more of a matter of perspective. We are not using `PATCH` much in the API where it would be legitimate IMO (like when updating the passphrase of metadata, i.e. when doing a partial update of a resource). Yet, we've stick to `PUT` here because people tend to be somewhat confused by the `PATCH` method. And, from the perspective of a sub-resource, a `PUT` is still acceptable for partially replacing an entity. But this is why we write:

```
PUT /wallets/{walletId}/passphrase
```

and not:

```
PUT /wallets/{walletId}

{ "passphrase:" ... }
```